### PR TITLE
squid:S1854 - Dead stores should be removed

### PR DESCRIPTION
**Pull request by georgekankava**
This pull request is focused on resolving occurrences of Sonar rule
squid:S1854 - Dead stores should be removed.
This pull request removes 120 minutes of technical debt.
You can find more information about the issue here:
https://dev.eclipse.org/sonar/rules/show/squid:S1854
Please let me know if you have any questions.
George Kankava

### CODE REVIEW

File: osmosis-apidb/src/main/java/org/openstreetmap/osmosis/apidb/v0_6/impl/ChangesetManager.java [hunk 1]
```diff
@@ -143,8 +143,7 @@ private void addChangeset(long changesetId, long userId) {
 
             // Insert the changeset tags.
             prmIndex = 1;
-            insertTagStatement.setLong(prmIndex++, changesetId);
-            insertTagStatement.setLong(prmIndex++, changesetId);
+            insertTagStatement.setLong(++prmIndex, changesetId);
```
Comment (georgekankava) [inline osmosis-apidb/src/main/java/org/openstreetmap/osmosis/apidb/v0_6/impl/ChangesetManager.java]: @rory Here I removed extra line which was doing the same and changed to ++prmIndex which first increments and then passes argument opposite to what was before.

File: osmosis-apidb/src/main/java/org/openstreetmap/osmosis/apidb/v0_6/impl/ChangesetManager.java [hunk 2]
```diff
@@ -143,8 +143,7 @@ private void addChangeset(long changesetId, long userId) {
 
             // Insert the changeset tags.
             prmIndex = 1;
-            insertTagStatement.setLong(prmIndex++, changesetId);
-            insertTagStatement.setLong(prmIndex++, changesetId);
```
Comment (brettch) [inline osmosis-apidb/src/main/java/org/openstreetmap/osmosis/apidb/v0_6/impl/ChangesetManager.java]: Good pickup.  I'm surprised this didn't cause an exception.

**Comment by brettch** [comment]
I like the idea of eliminating this "dead stores" issue.  I've noticed that IntelliJ is also complaining about this.  But I think there's a cleaner way to do it.

Removing the index variable post-increment from the last setXXX makes it inconsistent with previous lines.  If a new parameter gets added later it won't be immediately obvious that the previous line needs to be fixed.  How about we initialise the index variable to 0 and use pre-increment operator instead?

So, instead of:
prmIndex = 1;
insertXXXStatement.setXXX(prmIndex++, changesetId);
insertXXXStatement.setXXX(prmIndex++, changesetId);

How about we do:
prmIndex = 0;
insertXXXStatement.setXXX(++prmIndex, changesetId);
insertXXXStatement.setXXX(++prmIndex, changesetId);

That we all lines are consistent, and we avoid the dead code issue.

**Comment by rory** [comment]
@georgekankava Hi. You @mentioned me on this, but I think you made a mistake, I don't know anything about this. (Though as an active OSMer, you had me confused for a bit! :)) Maybe you meant to @mention someone else?

File: osmosis-apidb/src/main/java/org/openstreetmap/osmosis/apidb/v0_6/impl/ChangesetManager.java (hunk 1 above)
Comment (brettch) [inline osmosis-apidb/src/main/java/org/openstreetmap/osmosis/apidb/v0_6/impl/ChangesetManager.java]: This causes the app to fail.  prmIndex is now set to 2 when it should be one.  It causes the following error:
org.postgresql.util.PSQLException: No value specified for parameter 1.

**Comment by georgekankava** [comment]
@rory that's right. Sorry and thanks for letting me know :)

**Comment by georgekankava** [comment]
@brettch Updated.

**Comment by brettch** [comment]
@georgekankava, I see you've updated ChangesetManager, but not ApidbWriter.  Can you please update all classes to be consistent?

**Comment by georgekankava** [comment]
@brettch Sure. Updated. Please take a look.

File: osmosis-apidb/src/main/java/org/openstreetmap/osmosis/apidb/v0_6/ApidbWriter.java [hunk 3]
```diff
@@ -394,24 +394,22 @@ private void initialize() {
      * @param node The node containing the data to be inserted.
      */
     private void populateNodeParameters(PreparedStatement statement, int initialIndex, Node node) {
-        int prmIndex;
```
Comment (brettch) [inline osmosis-apidb/src/main/java/org/openstreetmap/osmosis/apidb/v0_6/ApidbWriter.java]: This is ignoring the initialIndex parameter which will cause prmIndex to be initialised incorrectly.  The initialIndex parameter can't be ignored.

File: osmosis-apidb/src/main/java/org/openstreetmap/osmosis/apidb/v0_6/ApidbWriter.java [hunk 4]
```diff
@@ -426,21 +424,19 @@ private void populateNodeParameters(PreparedStatement statement, int initialInde
      * @param way The way containing the data to be inserted.
      */
     private void populateWayParameters(PreparedStatement statement, int initialIndex, Way way) {
-        int prmIndex;
```
Comment (brettch) [inline osmosis-apidb/src/main/java/org/openstreetmap/osmosis/apidb/v0_6/ApidbWriter.java]: Same here, we can't ignore initialIndex.

Comment (georgekankava) [inline osmosis-apidb/src/main/java/org/openstreetmap/osmosis/apidb/v0_6/ApidbWriter.java]: @brettch Does this apply all occurrences or only this two?

**Comment by georgekankava** [comment]
@brettch Updated.

**Comment by brettch** [comment]
I'd suggest we close this PR.  The changes are broken in their current state and IMO aren't adding any real value.

**Pull request closed without merging**